storage/badger/protocol_state.go

```diff
@@ -48,13 +49,30 @@ func NewProtocolState(collector module.CacheMetrics,
 			return result, nil
 		}
 	}
+	cache := newCache[flow.Identifier, *flow.RichProtocolStateEntry](collector, metrics.ResourceProtocolState,
+		withLimit[flow.Identifier, *flow.RichProtocolStateEntry](cacheSize),
+		withStore(noopStore[flow.Identifier, *flow.RichProtocolStateEntry]),
+		withRetrieve(retrieve))
+
+	retrieveByBlockID := func(blockID flow.Identifier) func(tx *badger.Txn) (*flow.RichProtocolStateEntry, error) {
+		return func(tx *badger.Txn) (*flow.RichProtocolStateEntry, error) {
+			var protocolStateID flow.Identifier
+			err := operation.LookupProtocolState(blockID, &protocolStateID)(tx)
+			if err != nil {
+				return nil, fmt.Errorf("could not lookup identity table ID for block (%x): %w", blockID[:], err)
+			}
+			return cache.Get(protocolStateID)(tx)
```

Comment:
I have a minor preference for moving this second cache `Get` call outside of the `byBlockID` cache retrieval function (and into `ByBlockID`). That way each cache remains conceptually only a wrapper around one database call, rather than containing glue logic linking different database calls together. It's also easier to add things like a public `ProtocolStateIDByBlockID` method.

Reply:
Do you mean changing signature of secondary cache to `Cache[flow.Identifier, flow.Identifier]` which will map `block_id -> protocol_state_id`?

Reply:
> changing signature of secondary cache to Cache[flow.Identifier, flow.Identifier] which will map block_id -> protocol_state_id

Yeah, exactly